### ISSUE TITLE
Word-block count and table of contents popover accessibility

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3216]
Follow up to #2684. Thought there was already an issue open for this but couldn't find it. Feel free to close this if I've missed it.

![screen shot 2017-10-28 at 16 17 39](https://user-images.githubusercontent.com/1682452/32135215-9d217b0c-bbfb-11e7-80fa-57d6264f4c0b.png)

There are still some issues to address:
- placement of the "I" button as discussed on #2684 and #3166
- the toggle button "I" is empty and misses any accessible name. It used to have a visible "Info" text that was removed in https://github.com/WordPress/gutenberg/pull/2684/commits/7e3cc89b48de9d04daf4b86742010449890e784b without providing any alternate name to the button. As a consequence, there's nothing assistive technologies can announce other than "button"
- the toggle button should have an `aria-expanded` attribute
- it's an "icon-only" button, I wonder if it should use a tooltip as similar buttons do
- focus management: this is a general issue for the `Dropdown` (formerly "PopOver") component

**Comment by mtias** [comment]
@afercia do we have pending items here?

**Comment by afercia** [comment]
@mtias couple issues to consider:

1
When there are headings, opening the "Content Structure" move the initial focus to the first heading in the ToC. Not so ideal because all the previous content (the counts) get skipped.

2
Tabbing should be constrained within the Dropdown (formerly "PopOver") but I guess this should be addressed in that component. Is there already a specific issue for that?

I've noticed also a couple layout/usability potential problems, going to open a new issue.

**Comment by afercia** [comment]
Many things have changed here, I think this can be closed now.
- about constraining tabbing within the Popover, there's a specific issue see #5242 
- about initial focus when there are no headings, I'm going to push a PR

**Comment by mtias** [comment]
> about initial focus when there are no headings, I'm going to push a PR

Cool. I was thinking of a general solution as well where we can move focus to a popover but don't have to focus any specific item within it.

**Comment by afercia** [comment]
Hm the pattern recommended by the ARIA authoring practices is to always move focus to the first focusable element inside a popover or menu. We're discussing this in a separate issue, see #5559

**Comment by mtias** [comment]
We have gotten some feedback from users who don't expect a first item to be selected and confuse the focus ring with a heading style.

I think the prevalence of menus like these, that focus on the thing that opened the menu, contribute to that feeling:

![image](https://user-images.githubusercontent.com/548849/37303348-757e5988-262e-11e8-86a1-f8ff7a84db34.png)

**Comment by afercia** [comment]
Please refer to the ARIA resources posted on #5559. Moving focus to the first item is a requirement. We shouldn't try to solve what is a styling issue at the expenses of a required functionality.

**Comment by mtias** [comment]
I don't think it's a styling issue.
